商品詳細表示機能

fujiyasu0522 opened the pull request:
# What
商品の詳細ページの実装

# Why
商品の詳細を確認するため



ログインしていない時の詳細ページ
https://gyazo.com/53e2101b5855445e23f84fe965c34937

ログインしている時の詳細ページでかつ他人の出品商品の場合
https://gyazo.com/22a490e6b0dc37f8e72010c748604a9f

ログインしている時でかつ自身の出品商品の場合
https://gyazo.com/614af3d21e1559d6aaac0cecb162ecf0

fujiyasu0522 committed:
商品詳細表示機能

ログインしていない時の詳細ページ
https://gyazo.com/53e2101b5855445e23f84fe965c34937

ログインしている時の詳細ページでかつ他人の出品商品の場合
https://gyazo.com/22a490e6b0dc37f8e72010c748604a9f

ログインしている時でかつ自身の出品商品の場合
https://gyazo.com/614af3d21e1559d6aaac0cecb162ecf0

ken-1853 commented on app/views/items/show.html.erb:
```diff
@@ -19,51 +19,49 @@
         ¥ 999,999,999
```
```
     ¥ 999,999,999
```
こちら実際の商品の価格が表示されるようにしましょう！

ken-1853 commented on app/views/items/show.html.erb:
```diff
@@ -19,51 +19,49 @@
-    <p class="or-text">or</p>
-    <%= link_to "削除", "#", method: :delete, class:"item-destroy" %>
-
+    <% if user_signed_in? && current_user.id == @item.user_id %>
+      <%= link_to "商品の編集", "#", method: :get, class: "item-red-btn" %>
+      <p class="or-text">or</p>
+      <%= link_to "削除", "#", method: :delete, class:"item-destroy" %>
+    <% elsif user_signed_in? && current_user.id != @item.user_id %>
```
```
 <% if user_signed_in? && current_user.id == @item.user_id %>
<% elsif user_signed_in? && current_user.id != @item.user_id %>
```
trelloの要件通りの実装ができていますね！
是非こちらリファクタリングに挑戦してみましょう！

現状では、 user_signed_in?が繰り返し使用されています。
記述を工夫することで、条件式の記述を減らすことができます。
過去アプリのPicTweetなどを参考にしてみましょう！

ken-1853 commented on app/views/items/show.html.erb:
```diff
@@ -19,51 +19,49 @@
-
-
+    <% end %>
     <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```
```
 <%# //ログインしているユーザーと出品しているユーザーが、同一人物の場合と同一人物ではない場合で、処理を分けましょう %>
```
こちらのコメントアウトですが、不要であれば削除しましょう。
もし必要であれば、第三者でもわかるように「残す目的」「なぜ必要か」という理由などを、コメントとして書いておくことをおすすめいたします。

ここ以外もご自身で確認いただいて、不要なコメントアウトが残っていましたら、同様に削除しましょう！（未実装の箇所は残っていて構いません）

ken-1853 commented on app/views/items/show.html.erb:
```diff
@@ -19,51 +19,49 @@
+          <td class="detail-value"><%= @item.delivery_days.name %></td>
         </tr>
       </tbody>
     </table>
```
差分がないのでこちらに失礼します
103行目からの
```
 <%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
<a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
```
について、コメントアウトの指示に従ってカテゴリー名が表示されるようにしましょう！

ken-1853 reviewed (commented):
レビュー依頼ありがとうございます！
コメントいたしましたので、ご確認をお願いいたします。

修正頂いた後、コミットプッシュを確認いただき、再度レビュー依頼のご提出をお願い致します！

fujiyasu0522 committed:
修正1

ログインしていない時の詳細ページ
https://gyazo.com/dcb5832f3d01afbdec5108d1af25815e

ログインしている時の詳細ページでかつ他人の出品商品の場合
https://gyazo.com/e4b395ab1856b8f3ce2719c0ac30312a

ログインしている時でかつ自身の出品商品の場合
https://gyazo.com/a16958fb3ca1efe5bc2ce2b1c47de750

ken-1853 reviewed (commented):
レビュー依頼ありがとうございます！

ただ申し訳ございません、githubの反映処理が遅いだけかもしれないですが、こちらの修正分https://github.com/fujiyasu0522/furima-33357/pull/6/files#r645964375 がまだ反映されていないようです。
恐れ入りますが、commitやpush忘れなどをご確認いただき、再度プルリクエストをご提出いただけないでしょうか？

宜しくお願い致します

fujiyasu0522 committed:
修正（1）

Saka-toku commented on app/views/items/show.html.erb:
```diff
@@ -16,54 +16,52 @@
+          <td class="detail-value"><%= @item.delivery_days.name %></td>
         </tr>
       </tbody>
     </table>
```
https://github.com/fujiyasu0522/furima-33357/pull/6/files#r645964375
こちらが未修正のため修正をお願いいたします！

fujiyasu0522 commented:
修正（1）は修正1がgitへの反映がおそいため、再度コミットするために出しました。
修正内容は修正1を確認してください。

Saka-toku reviewed (commented):
レビュー依頼ありがとうございます！
修正1の内容とコミット後の内容を確認いたしました。
いまだ以下の内容が確認できないため、修正とコミットプッシュをお願いいたします。
```
103行目
<%# 詳細ページで表示されている商品のカテゴリー名を表示しましょう %>
<a href="#" class="another-item"><%= "商品のカテゴリー名" %>をもっと見る</a>
```

fujiyasu0522 committed:
修正2

ken-1853 commented:
レビュー依頼ありがとうございます。
修正点の確認ができました！
これにてLGTMとさせていただきます。実装お疲れさまでした！

念の為ご自身でもう一度挙動を確認いただき、問題なければマージしましょう！

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き頑張っていきましょう。応援しております！
